docs/features/software-catalog/life-of-an-entity.md

```diff
@@ -0,0 +1,127 @@
+passing them to the catalog builder in your backend initialization code, and you
+can easily write your own.
+
+An entity provider is a class that implements the `EntityProvider` interface. It
+has three main parts:
+
+- The identity: Each provider instance has a unique, stable identifier that the
+  database can use to keep track of the originator of each unprocessed entity.
+- The connection: During backend startup, each provider is attached to the
+  catalog runtime.
+- The stream of events: During its lifetime, the provider can issue change
+  events to the runtime at any point in time, to modify its set of unprocessed
+  entities.
+
+It is entirely up to the provider to choose how and when it produces these
+change events. For example, the app-config provider only fires off an update at
+startup and then lies dormant. The location database provider does an initial
+update at startup, and then small delta updates every time a location database
+change is detected. The LDAP provider is driven externally by a timer loop that
+occasionally triggers a full update. Some future provider may be entirely event
+driven, feeding off an event bus or web hook. There is no magic coordination
+among providers; if they need to arrange synchronization or locking among
+themselves for example to avoid duplicate work across multiple catalog service
+machines, they need to handle that out-of-band.
+
+The entities that are emitted get some coarse validation applied to them, to
+ensure that they adhere to the most basic rules of the system. They need to have
+a `kind`, a `metadata.name`, and optionally a `metadata.namespace`. Apart from
+that, the ingestion stage considers its work done, and stores the unprocessed
+entities to be picked up at a later time by the processing system.
+
+## Processing
+
+Every unprocessed entity comes with a timestamp, which tells at what time that
+the processing loop should next try to process it. When the entity first
+appears, this timestamp is set to "now" - asking for it to be picked up as soon
+as possible.
+
+Each catalog deployment has a number of processors installed. They are
+responsible for receiving unprocessed entities that the catalog decided are due
+for processing, and then running that data through a number of processing
+stages. mutating the entity and emitting auxiliary data about it. When all of
+that is done, the catalog takes all of that information and stores it as the
+processed entity, and errors and relations to other entities separately. Then,
+the catalog checks to see what entities are touched by that output, and issues
+stitching of those (see below).
+
+There are several stages involved in the processing.
+
+> TODO: More info here
+
+## Stitching
+
+The stitching is currently a fixed process, that cannot be modified or extended.
+This means that any modifications you want to make on the final result, has to
+happen during ingestion or processing.
+
```

Comment:
Perhaps adding something like this, edit as you wish:

Stitching finalises the entity by combining all data gathered such as relations, errors and the entity itself into a final object which is essentially the result from the catalog API. Entities that are no longer referenced by a location are annotated with `backstage.io/orphan: true` to indicate to humans and machines that this entity is no longer being updated.

Reply:
Good formulation, will steal it in whole or partially in the next update